server/src/proxyManagement.mjs

```diff
@@ -0,0 +1,191 @@
+/*
+* Copyright 2021 Comcast Cable Communications Management, LLC
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+* SPDX-License-Identifier: Apache-2.0
+*/
+
+'use strict';
+
+import promisify from 'util';
+import child_process from 'child_process';
+const exec = promisify.promisify(child_process.exec);
+import { parse } from 'url';
+import WebSocket from 'ws';
+let websocketConnection = null 
+
+async function initialize(receiverWSClient) {
+    const url = await buildWSUrl()
+    if (url) {
+      console.log("Establishing websocket connection")
+      let ws = new WebSocket(url)
+      return new Promise((res, rej) => {
+       
+        let websocket = ws
+        let openCallback = function(event) {
+          websocket.removeEventListener('close', openCallback)
+          websocket.removeEventListener('error', openCallback)
+          res(event.data)
+        }
+
+        ws.addEventListener('error', function(event) {
+          rej(event.data)
+        })
+
+        ws.addEventListener('close', function(event) {
+          rej(event.data)
+        })
+
+        ws.addEventListener('open', openCallback)
+
+        ws.onopen = function () {
+            console.log("connected to websocket client") 
+            websocketConnection = ws
+        }
+
+        ws.onclose = function(){
+            // connection closed, discard old websocket and create a new one in 5s
+            console.log("connection closed")
+            ws = null
+            websocketConnection = null
+            setTimeout(function() {
+                initialize(receiverWSClient);
+            }, 2000)
+        }
+
+        ws.onerror = function(err) {
+          console.log("connection error: ", err)
+        }
+
+        ws.on('message', function message(data) {
+            var buf = Buffer.from(data);
+            const oResponseMessage = {
+              jsonrpc: '2.0',
+              id: JSON.parse(buf.toString()).id,
+              ...JSON.parse(buf.toString())
+            };
+            const responseMessage = JSON.stringify(oResponseMessage);
+            receiverWSClient.send(responseMessage);
+            console.log(`Sent message to ripple: ${responseMessage} \n`);
```

Comment:
Can this be a bit more generic? Something like "Sent outbound message"

Reply:
Addressed